### PR TITLE
fix: move the current selector to fix scoping issues

### LINKED ISSUE #1711: Scoping SCSS under a different body class breaks some components
To preview experimental carbon in `carbon-components-angular` we generate two `preview.scss` files and import carbon under two different containing classes (`.carbon` and `.experimental`).

As expected some tweaks are required to make this work (manually scoping the reset and fonts for example) ... but strangely enough some component styles also break (mainly padding/margins) ... [see these lines for what I mean](https://github.com/IBM/carbon-components-angular/blob/c8d4f381905379c91758ca2bb05fd4ba90133e4a/.storybook/preview.scss#L24-L58). It doesn't appear to be an issue for X, but it's definitely unexpected behavior.

### PR DESCRIPTION
**Pull request by cal-smith**
Closes #1711 for all instances that I'm presently aware of

TL;DR from the issue:

If `&` is used as a second level selector, it inherits the _entire_  first level selector. So `.#{$prefix}--structured-list-row` becomes `.carbon .bx--structured-list-row` and 

```sass
.#{$prefix}--structured-list--selection &:hover:not(.#{$prefix}--structured-list-row--header-row)
``` 

becomes 
```sass
.bx--structured-list--selection .carbon .bx--structured-list-row:hover:not(.bx--structured-list-row--header-row)
```

instead of
```sass
.carbon .bx--structured-list--selection .bx--structured-list-row:hover:not(.bx--structured-list-row--header-row)
```

This PR moves a few of the selectors around. This does result in a few styles moving away from their direct container, but it does express the intent of the selectors better.

As far as I can tell none of the X styles are affected by this.

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @cal-smith!

**Review by lovemecomputer — APPROVED** [review]
awesome thanks for catching this!
